cve_bin_tool/strings.py

```diff
@@ -6,31 +6,31 @@
 and MacOS.
 """
 
-import sys
 import string
+import sys
 
-try:
-    import cve_bin_tool.pstring as pstring
-except ImportError:
-    pstring = None
 
+class Strings:
+    # printable characters
+    PRINTABLE = set(range(32, 128))
+    # add tab to the printable character
+    PRINTABLE.add(9)
 
-class Strings(object):
     def __init__(self, filename=""):
         self.filename = filename
-        self.output = ""
-        self.printable = set(string.printable)
+        self.output = [""]
 
     def parse(self):
-        # Use c extention if available
-        if pstring is not None:
-            return pstring.string(self.filename)
-        else:
-            return self.parse_3()
-
-    def parse_3(self):
-        with open(self.filename, "r", encoding="ascii", errors="surrogateescape") as f:
-            self.output = "".join(filter(lambda x: x in self.printable, f.read()))
+        with open(self.filename, "rb") as f:
```

Comment:
Have we re-benchmarked with this for loop based implementation?

Reply:
Yes, the benchmark I have shared is of this for loop implementation which is way faster than other implementation.

Reply:
Let me put that benchmark code again share the results.